browser/ui/webui/brave_education/education_page_ui.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// based on //chrome/browser/ui/webui/whats_new/whats_new_ui.cc
+
+#include "brave/browser/ui/webui/brave_education/education_page_ui.h"
+
+#include <string>
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_browser_command/brave_browser_command_handler.h"
+#include "brave/browser/ui/webui/brave_education/brave_education_handler.h"
+#include "brave/browser/ui/webui/brave_education/brave_education_page_delegate_desktop.h"
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/components/brave_education/education_urls.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/grit/brave_education_resources.h"
+#include "brave/grit/brave_education_resources_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "chrome/grit/branded_strings.h"
+#include "chrome/grit/generated_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+
+namespace {
+
+void CreateAndAddWhatsNewUIHtmlSource(content::WebUI* web_ui,
+                                      Profile* profile) {
+  content::WebUIDataSource* source =
+      content::WebUIDataSource::CreateAndAdd(profile, kBraveGettingStartedHost);
+  webui::SetupWebUIDataSource(
+      source, base::span<const webui::ResourcePath>(kBraveEducationResources),
+      IDR_BRAVE_EDUCATION_BRAVE_EDUCATION_HTML);
+
+  AddBackgroundColorToSource(source, web_ui->GetWebContents());
+
+  static constexpr webui::LocalizedString kStrings[] = {
+      {"title", IDS_WELCOME_HEADER}};
+  source->AddLocalizedStrings(kStrings);
+
+  // Allow embedding of iframe content from allowed domains.
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ChildSrc,
+      "child-src chrome://webui-test https://brave.com/ "
```

Comment:
```suggestion
      "child-src chrome://webui-test https://browser-education.brave.com "
```

Reply:
per other comment the brave url should be a constant